internal/pkg/controller/controller.go

```diff
@@ -0,0 +1,332 @@
+		return
+	}
+
+	for i := 0; i < threadiness; i++ {
+		go wait.Until(c.runWorker, time.Second, stopCh)
+	}
+
+	<-stopCh
+	logrus.Infof("Stopping Controller for type ", c.resource)
+}
+
+func (c *Controller) runWorker() {
+	for c.processNextItem() {
+	}
+}
+
+func (c *Controller) processNextItem() bool {
+	// Wait until there is a new item in the working queue
+	event, quit := c.queue.Get()
+	if quit {
+		return false
+	}
+	// Tell the queue that we are done with processing this key. This unblocks the key for other workers
+	// This allows safe parallel processing because two events with the same key are never processed in
+	// parallel.
+	defer c.queue.Done(event)
+
+	// Invoke the method containing the business logic
+	err := c.takeAction(event.(Event))
+	// Handle the error if something went wrong during the execution of the business logic
+	c.handleErr(err, event)
+	return true
+}
+
+// main business logic that acts bassed on the event or key
+func (c *Controller) takeAction(event Event) error {
+
+	obj, _, err := c.indexer.GetByKey(event.key)
+	if err != nil {
+		logrus.Infof("Fetching object with key %s from store failed with %v", event.key, err)
+	}
+	if obj == nil {
+		logrus.Infof("Error in Action")
+	} else {
+		logrus.Infof("Detected changes in object %s", obj)
+		// process events based on its type
+		logrus.Infof("Performing '%s' action for controller of type '%s'", event.eventType, c.resource)
+		switch event.eventType {
+			case "create":
+				ObjectCreated(obj, c.client)
+			case "update":
+				ObjectUpdated(obj, c.client)
+			case "delete":
+				ObjectDeleted(obj)
+		}
+	}
+	return nil
+}
+
+// handleErr checks if an error happened and makes sure we will retry later.
+func (c *Controller) handleErr(err error, key interface{}) {
+	if err == nil {
+		// Forget about the #AddRateLimited history of the key on every successful synchronization.
+		// This ensures that future processing of updates for this key is not delayed because of
+		// an outdated error history.
+		c.queue.Forget(key)
+		return
+	}
+
+	// This controller retries 5 times if something goes wrong. After that, it stops trying.
+	if c.queue.NumRequeues(key) < 5 {
+		logrus.Infof("Error syncing events %v: %v", key, err)
+
+		// Re-enqueue the key rate limited. Based on the rate limiter on the
+		// queue and the re-enqueue history, the key will be processed later again.
+		c.queue.AddRateLimited(key)
+		return
+	}
+
+	c.queue.Forget(key)
+	// Report to an external entity that, even after several retries, we could not successfully process this key
+	runtime.HandleError(err)
+	logrus.Infof("Dropping the key %q out of the queue: %v", key, err)
+}
+
+// ObjectCreated Do nothing for default handler
+func ObjectCreated(obj interface{}, client kubernetes.Interface) {
+	message := "Configmap: `" + obj.(*v1.ConfigMap).Name + "`has been created in Namespace: `" + obj.(*v1.ConfigMap).Namespace + "`"
+	logrus.Infof(message)
+	err := rollingUpgradeDeployments(obj, client)
```

Comment:
This shouldn't be hardcoded. Figure out a way to detect which resources should be updated by using the annotation